src/api/app/queries/notifications_finder.rb

```diff
@@ -44,4 +44,10 @@ def for_notifiable_type(type = 'unread')
   def for_project_name(project_name)
     unread.joins(:projects).where(projects: { name: project_name })
   end
+
+  def current_notification(notification_id)
+    return unless User.session && notification_id
+
+    self.class.new.for_subscribed_user.where(id: notification_id).first
```

Comment:
I suggest using `find_by(...)` instead of `where(...).first`. They differ a bit since `first` orders the records, then return the first one. As for `find_by`, it returns the first record without ordering the records. In our case, we don't need to order the records since there's only one possible notification with the ID we're looking for. We could use `find`, but it will raise an error if a notification cannot be found for that ID. It might happen if someone is trying to be funny and pass us some garbage.

The subtle differences between the methods are described [here](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html).

```suggestion
    self.class.new.for_subscribed_user.find_by(id: notification_id)
```

Reply:
Changed, thanks!